stream-api/src/main/kotlin/fund/cyber/markets/api/common/BroadcastersIndex.kt

```diff
@@ -0,0 +1,64 @@
+package fund.cyber.markets.api.common
+
+import fund.cyber.markets.api.orders.OrdersBroadcaster
+import fund.cyber.markets.api.trades.TradesBroadcaster
+import fund.cyber.markets.model.Order
+import fund.cyber.markets.model.TokensPair
+import fund.cyber.markets.model.Trade
+import kotlinx.coroutines.experimental.channels.Channel
+import java.util.LinkedList
+import java.util.concurrent.ConcurrentHashMap
+
+class TradesBroadcastersIndex : BroadcastersIndex<Trade>() {
+    override fun newChannel(exchange: String, pair: TokensPair, channel: Channel<Trade>) {
+        val broadcaster = TradesBroadcaster(channel)
+        index.getOrPut(pair, { ConcurrentHashMap() }).put(exchange, broadcaster)
+    }
+}
+
+class OrdersBroadcastersIndex : BroadcastersIndex<List<Order>>() {
+    override fun newChannel(exchange: String, pair: TokensPair, channel: Channel<List<Order>>) {
+        val broadcaster = OrdersBroadcaster(channel)
+        index.getOrPut(pair, { ConcurrentHashMap() }).put(exchange, broadcaster)
+    }
+}
+
+abstract class BroadcastersIndex<T> : ChannelsIndexUpdateListener<T> {
+
+    protected val index: MutableMap<TokensPair, MutableMap<String, Broadcaster>> = ConcurrentHashMap()
+
+    fun broadcastersFor(pairs: List<TokensPair>, exchanges: List<String>): Collection<Broadcaster> {
+        return when {
```

Comment:
This `when` statement looks too complicated and not readable, create functions with branchs bodies with good names.
Also use `.isNotEmpty()` is available here.